projects/observability/src/pages/saved-queries/saved-queries.component.ts

```diff
@@ -0,0 +1,52 @@
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { NavigationService, PreferenceService } from '@hypertrace/common';
+import { DrilldownFilter, ExplorerService } from '../explorer/explorer-service';
+import { SavedQuery } from '../explorer/explorer.component';
+
+@Component({
+  styleUrls: ['./saved-queries.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="saved-queries">
+      <ht-page-header class="explorer-header"></ht-page-header>
+      <div class="query-list-container">
+        <a *ngFor="let query of savedQueries" (click)="onClick(query)">
+          <span *ngFor="let filter of query.filters">{{ filter.userString }}</span>
+        </a>
+      </div>
+      <p class="not-found-text" *ngIf="savedQueries.length === 0">
+        You haven't saved any queries! Go to Explorer page to save a query.
+      </p>
+    </div>
+  `
+})
+export class SavedQueriesComponent implements OnDestroy {
+  public savedQueries: SavedQuery[] = [];
+  private readonly subscriptions: Subscription = new Subscription();
+
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly preferenceService: PreferenceService,
+    private readonly explorerService: ExplorerService
+  ) {
+    this.subscriptions.add(
+      this.preferenceService.get('savedQueries', []).subscribe(queries => {
+        this.savedQueries = queries as SavedQuery[];
+      })
+    );
+  }
+
+  public ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
+  }
+
+  public onClick(query: SavedQuery): void {
+    this.subscriptions.add(
+      this.explorerService
+        .buildNavParamsWithFilters(query.scopeQueryParam, query.filters as DrilldownFilter[])
+        .subscribe(navParams => this.navigationService.navigate(navParams))
+    );
```

Comment:
Just thinking if we can do something different here. Not able to objectively tell what can be better.
Tried a variation like below.

```typescript
  public async onClick(query: SavedQuery): Promise<void> {
     this.navigationService.navigate(await this.explorerService
        .buildNavParamsWithFilters(query.scopeQueryParam, query.filters as DrilldownFilter[])
        .pipe(first()).toPromise());
  }
```

Reply:
So, found a workaround, you can use the following. If this makes sense to you, you can give it a shot. 
```
<ht-link paramsOrUrl="getExplorerNavigationParams(query)">
{...clickable query box code without anchor tag}
</ht-link>
```

Reply:
Yeah, it looks weird having to call subscribe on every click. But I wasn't sure how to call a method on every click that returns an observable. Let me try the `<ht-link>` approach.

Reply:
Looks like it worked. Changes pushed ✅